pkg/kubelet/status/manager.go

```diff
@@ -323,8 +323,14 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	// TODO: make me easier to express from client code
 	pod, err := m.kubeClient.Pods(status.podNamespace).Get(status.podName)
 	if errors.IsNotFound(err) {
-		glog.V(3).Infof("Pod %q (%s) was deleted on the server", status.podName, uid)
-		m.deletePodStatus(uid)
+		glog.V(3).Infof("Pod %q (%s) does not exist on the server", status.podName, uid)
+		// Note that we do not delete the pod status because the pod to update may
+		// be a static pod whose mirror pod is in the process of recreation.
+		// Deleting the status without clearing the corresponding
+		// apiStatusVersions entry may lead to wrong versioning and stale status.
```

Comment:
Yes, resolving #8818 would resolve this issue too. However, without having external component mucking the status, I would expect kubelet to have correct versioning anyway. As for having proper version, do you mean to record the resourceVersion of the Pod object as I mentioned earlier or just the approach we decided, as stated in https://github.com/kubernetes/kubernetes/issues/8818#issuecomment-161750940?

Regardless, I think the PR is valid in the sense that the condition in the code does not apply to static pod at all. Do you want me to fix the comment? I don't think I can describe the case concisely but I could try.